src/software/pipeline/main_panoramaWarping.cpp

```diff
@@ -1094,8 +1101,9 @@ int aliceVision_main(int argc, char **argv)
   // If panorama width is undefined, estimate it
   if (panoramaSize.first <= 0)
   {
+    float ratioUpscale = float(percentUpscale) / 100.0f;
     std::pair<int, int> optimalPanoramaSize;
-    if (computeOptimalPanoramaSize(optimalPanoramaSize, sfmData))
+    if (computeOptimalPanoramaSize(optimalPanoramaSize, sfmData, ratioUpscale))
     {
       panoramaSize = optimalPanoramaSize;
     }
```

Comment:
If computeOptimalPanoramaSize fails, we should probably throw. Else panoramaSize.first will remain <= 0.